onnxruntime/core/platform/posix/env.cc

```diff
@@ -203,13 +217,27 @@ class PosixThread : public EnvThread {
 
  private:
   static void* ThreadMain(void* param) {
-    std::unique_ptr<Param> p((Param*)param);
+    std::unique_ptr<Param> p(static_cast<Param*>(param));
     ORT_TRY {
+#if !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_AIX)
+      if (p->affinity_mask != std::numeric_limits<size_t>::max()) {
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        CPU_SET(p->affinity_mask, &cpuset);
+        // pthread_setaffinity_np() does not set errno, it returns it.
+        auto ret = pthread_setaffinity_np(pthread_self(), sizeof(cpu_set_t), &cpuset);
+        if (ret != 0) {
+          auto [err_no, err_msg] = GetSystemError(ret);
+          LOGS_DEFAULT(ERROR) << "pthread_setaffinity_np failed for thread: " << gettid()
+                              << ", error code: ", err_no, " error msg: ", err_msg;
```

Comment:
>", e [](http://example.com/codeflow?start=48&length=4)

the skipping of "<<" between chars and int is intentional?

Reply:
Unntentional